parseAndPopulate/modulesComplicatedAlgorithms.py

```diff
@@ -792,95 +772,84 @@ def __parse_semver(self):
+                                            upgraded_version = '{}.{}.{}'.format(versions[0], versions[1], ver)
+                                            modules[x]['semver'] = upgraded_version
+                                            response = data['{}@{}'.format(mod['name'], mod['revision'])]
+                                            response['derived-semantic-version'] = upgraded_version
+                                            name_revision = '{}@{}'.format(response['name'], response['revision'])
+                                            if self.__new_modules.get(name_revision) is None:
+                                                self.__new_modules[name_revision] = response
+                                            else:
+                                                self.__new_modules[name_revision]['derived-semantic-version'] = response[
+                                                    'derived-semantic-version']
                                         else:
-                                            self.__new_modules[name_revision]['derived-semantic-version'] = response[
-                                                'derived-semantic-version']
+                                            versions = modules[x - 1]['semver'].split('.')
+                                            ver = int(versions[1])
+                                            ver += 1
+                                            upgraded_version = '{}.{}.{}'.format(versions[0], ver, 0)
+                                            modules[x]['semver'] = upgraded_version
+                                            response = data['{}@{}'.format(mod['name'], mod['revision'])]
+                                            response['derived-semantic-version'] = upgraded_version
+                                            name_revision = '{}@{}'.format(response['name'], response['revision'])
+                                            if self.__new_modules.get(name_revision) is None:
+                                                self.__new_modules[name_revision] = response
+                                            else:
+                                                self.__new_modules[name_revision]['derived-semantic-version'] = response[
+                                                    'derived-semantic-version']
                                     else:
                                         versions = modules[x - 1]['semver'].split('.')
-                                        ver = int(versions[1])
+                                        ver = int(versions[0])
                                         ver += 1
-                                        upgraded_version = '{}.{}.{}'.format(versions[0], ver, 0)
+                                        upgraded_version = '{}.{}.{}'.format(ver, 0, 0)
                                         modules[x]['semver'] = upgraded_version
```

Comment:
**NOTE**: This was not changed, two identical blocks of code were merged below after rewriting the if-else condition, and so git probably evaluated it as a change